_tools/riot-boards.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+# Parse RIOT boards and dump them in YAML format
+#
+# Usage:
+#  $ ./riot-boards.sh repo
+# * repo: path to RIOT's repository
+
+RIOTBASE=$1
+
+parseboards () {
+    cd $RIOTBASE
+    grep -R --include doc.txt defgroup ./boards | grep -v "common" | grep -v "config" | \
+    awk '/boards_/ {match($0, "(boards_[a-zA-Z0-9\\-_]+)\\s+(.*)$",a); file = a[1] ; board = a[2]; url = "https://doc.riot-os.org/group__"; gsub("_", "__", file); print "- name: " board "\n  apiurl: " url file ".html"}'
+    cd - > /dev/null
```

Comment:
```suggestion
    grep -R --include doc.txt defgroup ${RIOTBASE}/boards | grep -v "common" | grep -v "config" | \
    awk '/boards_/ {match($0, "(boards_[a-zA-Z0-9\\-_]+)\\s+(.*)$",a); file = a[1] ; board = a[2]; url = "https://doc.riot-os.org/group__"; gsub("_", "__", file); print "- name: " board "\n  apiurl: " url file ".html"}'
```

TBH all these awk commands give a headache. Is there a way to merge all these scripts in a single python script ? It would be IMO more readable (since Python is quite good for processing strings).

Reply:
in contrast to Python, shell scripting, including awk, runs out of the box. i'm in favor of using basic tools.

Reply:
I'm also in favor of basic tools (like Python) but readability matters and here (and more below), it's hard to read. A Python script would give a more readable logic and give more control.

Reply:
I rewrote the files to make them more scripts rather than one liners. I think it's much easier to follow now.
Is it ok?

Reply:
the beauty of one-liners for this trivial stuff dissolved ;). ok, i can live with this. thanks for the effort!